web/app/js/components/ResourceDetail.jsx

```diff
@@ -53,67 +47,131 @@ export class ResourceDetailBase extends React.Component {
     return {
       namespace: resource.namespace,
       resourceName: resource.name,
-      resourceType: resource.type
+      resourceType: resource.type,
+      resource,
+      pollingInterval: 2000,
+      resourceMetrics: [],
+      podMetrics: [],
+      pendingRequests: false,
+      loaded: false,
+      error: null
     };
   }
 
-  banner = () => {
-    const {error} = this.props;
+  componentDidMount() {
+    this.loadFromServer();
+    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+  }
 
-    if (!error) {
+  componentWillReceiveProps(newProps) {
```

Comment:
Saw this note in the React docs:
> If you used componentWillReceiveProps for re-computing some data only when a prop changes, use a memoization helper instead.

How does this relate to get the initial prop state for the component? Do we need to worry about this? Asking becuase I don't have a clear idea of how the `componentWillReceiveProps` work.

Reply:
ah sorry forgot to comment about this before merging. iirc the reason this was put in here was because we weren't updating the page as soon as the user navigated to another page (there was some delay where the old props/page still showed). Here, it's not that we're recomputing data, but mostly just clearing the page's current state/metrics/display that relate to the props.

I think we could maybe switch from using this to getDerivedStateFromProps and that would also fix this problem. We can look into it!